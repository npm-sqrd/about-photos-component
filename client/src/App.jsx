import React from 'react';
import { render } from 'react-dom';
import ajax from './ajax.js';
import sampleData from '../../data/sampleData.js';
import PropTypes from 'prop-types';
import About from './About.jsx';
import SideBar from './SideBar.jsx';
import Banner from './Banner.jsx';

class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      id: props.data.id,
      name: props.data.name,
      description: props.data.about.description,
      hours: props.data.about.hours,
      phone: props.data.about.phone,
      price: props.data.about.price,
      style: props.data.about.style,
      banner: props.data.banner,
      photo: props.data.photo,
    }
  }

  componentDidMount() {
    this.fetch();
  }

  fetch() {
    ajax.get((err, data) => {
      if (err) {
        throw err;
      } else {
        this.setState({
          restaurant: data.restaurant,
        })
      }
    });
  }

  render() {
    return (
      <div>
        <div><Banner banner={this.state.banner} /></div>
          <div><About description={this.state.description} name={this.state.name} photo={this.state.photo} /></div>
        <div><SideBar hours={this.state.hours} phone={this.state.phone} price={this.state.price} style={this.state.style} /></div>
      </div>
    );
  }
}

render(<App data={sampleData[0]}/>, document.getElementById('app'));

export default App;
