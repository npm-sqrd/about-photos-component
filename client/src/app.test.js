import React from 'react';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import App from './index.jsx';

Enzyme.configure({ adapter: new Adapter() });

describe('App Component', () => {
  test('should render correctly', () => {
    expect(Enzyme.shallow(<App />)).toMatchSnapshot();
  });
});
