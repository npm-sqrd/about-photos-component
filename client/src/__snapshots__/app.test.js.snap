// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <About
        data={
          Object {
            "about": Object {
              "description": "Est dignissimos voluptas cumque. In enim cum est eligendi. Ratione reprehenderit sequi saepe nulla repellendus omnis.
 
Dolorum non error. Magni unde ratione sunt cumque impedit tempore temporibus. Saepe voluptas deserunt illum quia voluptate laborum molestiae cumque sint. Eius error nulla quas. Porro sed libero. Eum voluptatum totam ut nulla mollitia vel voluptates.
 
Voluptas rem occaecati aliquid voluptatem soluta dolorem cum ut. Ut mollitia eos quisquam. Omnis tempora similique.",
              "hours": "M - F, 7am - 8pm",
              "phone": "041-300-0703",
              "price": "$$",
              "style": "dress-code enforced",
            },
            "banner": "https://s3-us-west-1.amazonaws.com/imagesfornamson/int5.jpeg",
            "id": 90976,
            "name": "Beard Papa's",
            "photo": Array [
              "https://s3-us-west-1.amazonaws.com/imagesfornamson/icecream1.jpeg",
              "https://s3-us-west-1.amazonaws.com/imagesfornamson/icecream2.jpeg",
              "https://s3-us-west-1.amazonaws.com/imagesfornamson/icecream3.jpeg",
            ],
          }
        }
      />,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "data": Object {
          "about": Object {
            "description": "Est dignissimos voluptas cumque. In enim cum est eligendi. Ratione reprehenderit sequi saepe nulla repellendus omnis.
 
Dolorum non error. Magni unde ratione sunt cumque impedit tempore temporibus. Saepe voluptas deserunt illum quia voluptate laborum molestiae cumque sint. Eius error nulla quas. Porro sed libero. Eum voluptatum totam ut nulla mollitia vel voluptates.
 
Voluptas rem occaecati aliquid voluptatem soluta dolorem cum ut. Ut mollitia eos quisquam. Omnis tempora similique.",
            "hours": "M - F, 7am - 8pm",
            "phone": "041-300-0703",
            "price": "$$",
            "style": "dress-code enforced",
          },
          "banner": "https://s3-us-west-1.amazonaws.com/imagesfornamson/int5.jpeg",
          "id": 90976,
          "name": "Beard Papa's",
          "photo": Array [
            "https://s3-us-west-1.amazonaws.com/imagesfornamson/icecream1.jpeg",
            "https://s3-us-west-1.amazonaws.com/imagesfornamson/icecream2.jpeg",
            "https://s3-us-west-1.amazonaws.com/imagesfornamson/icecream3.jpeg",
          ],
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <About
          data={
            Object {
              "about": Object {
                "description": "Est dignissimos voluptas cumque. In enim cum est eligendi. Ratione reprehenderit sequi saepe nulla repellendus omnis.
 
Dolorum non error. Magni unde ratione sunt cumque impedit tempore temporibus. Saepe voluptas deserunt illum quia voluptate laborum molestiae cumque sint. Eius error nulla quas. Porro sed libero. Eum voluptatum totam ut nulla mollitia vel voluptates.
 
Voluptas rem occaecati aliquid voluptatem soluta dolorem cum ut. Ut mollitia eos quisquam. Omnis tempora similique.",
                "hours": "M - F, 7am - 8pm",
                "phone": "041-300-0703",
                "price": "$$",
                "style": "dress-code enforced",
              },
              "banner": "https://s3-us-west-1.amazonaws.com/imagesfornamson/int5.jpeg",
              "id": 90976,
              "name": "Beard Papa's",
              "photo": Array [
                "https://s3-us-west-1.amazonaws.com/imagesfornamson/icecream1.jpeg",
                "https://s3-us-west-1.amazonaws.com/imagesfornamson/icecream2.jpeg",
                "https://s3-us-west-1.amazonaws.com/imagesfornamson/icecream3.jpeg",
              ],
            }
          }
        />,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "data": Object {
            "about": Object {
              "description": "Est dignissimos voluptas cumque. In enim cum est eligendi. Ratione reprehenderit sequi saepe nulla repellendus omnis.
 
Dolorum non error. Magni unde ratione sunt cumque impedit tempore temporibus. Saepe voluptas deserunt illum quia voluptate laborum molestiae cumque sint. Eius error nulla quas. Porro sed libero. Eum voluptatum totam ut nulla mollitia vel voluptates.
 
Voluptas rem occaecati aliquid voluptatem soluta dolorem cum ut. Ut mollitia eos quisquam. Omnis tempora similique.",
              "hours": "M - F, 7am - 8pm",
              "phone": "041-300-0703",
              "price": "$$",
              "style": "dress-code enforced",
            },
            "banner": "https://s3-us-west-1.amazonaws.com/imagesfornamson/int5.jpeg",
            "id": 90976,
            "name": "Beard Papa's",
            "photo": Array [
              "https://s3-us-west-1.amazonaws.com/imagesfornamson/icecream1.jpeg",
              "https://s3-us-west-1.amazonaws.com/imagesfornamson/icecream2.jpeg",
              "https://s3-us-west-1.amazonaws.com/imagesfornamson/icecream3.jpeg",
            ],
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
